dolphin/drivers/helper.py

```diff
@@ -25,3 +27,57 @@ def get_access_info(context, storage_id):
     access_info_dict = access_info.to_dict()
     access_info_dict['password'] = cryptor.decode(access_info_dict['password'])
     return access_info_dict
+
+
+def create_access_info(context, access_info):
+    access_info['password'] = cryptor.encode(access_info['password'])
+    return db.access_info_create(context, access_info)
+
+
+def update_access_info(context, storage_id, access_info):
+    return db.access_info_update(context,
+                                 storage_id,
+                                 access_info)
+
+
+def create_storage(context, storage):
+    return db.storage_create(context, storage)
+
+
+def update_storage(context, storage_id, storage):
+    return db.storage_update(context, storage_id, storage)
+
+
+def check_storage_exist(context, storage):
+    if not storage:
+        msg = _("Storage could not be found.")
+        raise exception.StorageNotFound(message=msg)
+
+    if not storage.get('serial_number'):
+        msg = _("Serial number should be provided by storage.")
+        raise exception.InvalidResults(message=msg)
+
+    filters = dict(serial_number=storage['serial_number'])
+    storage_list = db.storage_get_all(context, filters=filters)
+    if storage_list:
+        msg = (_("Failed to register storage. Reason: same serial number: "
+                 "%s detected.") % storage['serial_number'])
+        LOG.error(msg)
+        raise exception.InvalidRequest(msg)
+
+
+def check_storage_with_access_info(context, storage_id, storage_new):
```

Comment:
suggesting  some another name fr this function
match_storage_serial_number()
or
validate_storage()

Reply:
done, plz check